main.c

```diff
@@ -146,10 +196,8 @@ main(int argc, char *argv[])
 		}
 
 		showLogo();
-		if (stop_ms)
-			usleep(stop_ms * 1000);
-		else
-			select(0, NULL, NULL, NULL, NULL);
+		set_timespec(&time, stop_ms);
+		wait_signalfd(sigfd, stop_ms ? &time : NULL);
```

Comment:
Hmm, thinking out loud mainly here: I used pselect() mostly because select() doesn't preserve timeval values. Then again it's not much work to set the values on every iteration and it's not like this is some time critical piece of code [and it's already zeroing fs_set on every iteration, I'm not sure which one takes actually more time] so it'd not be that much of a extra cost. 

The only thing it makes impossible is to use zero timeout (as we need to be able to specify NULL with 0) but I think that's probably not needed in this program and if someone some day needs it, they can just adjust. Ok, I'll change it.